### PR TITLE
Build dev commit for Tensorflow 2.8.0

### PR DESCRIPTION
**Pull request by xhochy**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -89,7 +91,11 @@ outputs:
         - libcxx >={{ cxx_compiler_version }}.a0  # [osx]
     test:
       commands:
+        {% if git_commit != "" %}
+        - test -f $PREFIX/lib/libLLVM-{{ major_ver }}git${SHLIB_EXT}  # [unix]
```
Comment (hmaarrfk) [inline recipe/meta.yaml:95]: So you want the git suffix?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,24 +1,26 @@
-{% set version = "14.0.0.rc1" %}
+{% set version = "14.0.0.dev55c71c9eac9b" %}
```
Comment (hmaarrfk) [inline recipe/meta.yaml:1]: Maybe we can choose something that sorts more nicely?

Comment (xhochy) [reply recipe/meta.yaml:1]: The output of `git describe` is sadly not the best for this commit: `llvmorg-14-init-12920-g55c71c9eac9b`, we can use the distance from the branching tag but that didn't seem to add any value imho.

File: recipe/meta.yaml (hunk 1 above)
Comment (xhochy) [reply recipe/meta.yaml:95]: For builds from git, this is expected. It is also a good indicator that this shouldn't show in any downstream packages.

File: recipe/meta.yaml (hunk 2 above)
Comment (h-vetinari) [reply recipe/meta.yaml:1]: Can we represent the tensorflow version that this commit comes from in the version (or at least) build string?

E.g. `14.0.0.tf280.dev...`

File: recipe/conda_build_config.yaml [hunk 3]
```diff
@@ -5,7 +5,7 @@ cxx_compiler:          # [win]
 MACOSX_SDK_VERSION:    # [osx and x86_64]
   - 10.12              # [osx and x86_64]
 channel_targets:
-  - conda-forge llvm_rc
+  - conda-forge llvm_dev
```
Comment (h-vetinari) [inline recipe/conda_build_config.yaml:8]: I'd prefer a dedicated label (and branch...?) `llvm_tf`

Comment (h-vetinari) [reply recipe/conda_build_config.yaml:8]: In the sense that it's not a dev-package (i.e. not meant for consumption). If people were to install a new tensorflow and get packages from `conda-forge/label/llvm_dev` that feels much more iffy than `conda-forge/label/llvm_dev`.

Comment (xhochy) [reply recipe/conda_build_config.yaml:8]: These packages are only installed during the build. As we link statically, end-users will never see them

File: recipe/meta.yaml (hunk 2 above)
Comment (xhochy) [reply recipe/meta.yaml:1]: I would also be OK with `14.0.0.tf280_55c71c9eac9b`. Personally, I would like to have the git commit in the version to make its origin really obvious.

**Comment by xhochy** [comment]
@h-vetinari Would pushing this as `14.0.0.tf280_55c71c9eac9b` to an `llvm_tf` label suffice to merge here or are there other open points?

**Comment by h-vetinari** [comment]
No blocker from my side. I'd prefer the `llvm_tf` label in any case, but if the binaries will only be linked statically, not even that is necessary.

**Review by h-vetinari — APPROVED** [review]
Small nits regarding build string and channel label (see above) are optional.

**Pull request merged**